lib/puppet/util/selinux.rb

```diff
@@ -137,6 +142,33 @@ def set_selinux_default_context(file)
     nil
   end
 
+  ##
+  # normalize_selinux_category is an internal, method that converts
+  # all selinux categories to the human-readable representation,
+  # avoiding potential issues when mcstransd is not functional.
+  #
+  # It is not marked private because it is needed by File's
+  # selcontext.rb, but it is not intended for use outside of Puppet's
+  # code.
+  #
+  # @return [String] the human-readable category name
+  def normalize_selinux_category(category)
+    # We don't cache this, but there's already a ton of duplicate work
+    # in the selinux handling code.
+
+    File.open(Selinux.selinux_translations_path).each do |line|
```

Comment:
My experience with "a file exists always" is mostly that its not true. although I couldn't imagine that it ever wouldn't exist before there was the one important case where it didn't exist. ;-)